app/src/main/java/org/mozilla/focus/home/TopSiteAdapter.java

```diff
@@ -81,19 +78,16 @@ public void onBindViewHolder(SiteViewHolder holder, int position) {
         holder.text.setText(site.getTitle());
         String favIconUri = site.getFavIconUri();
         if (favIconUri != null) {
-            Glide.with(holder.img.getContext())
-                    .asBitmap()
-                    .load(favIconUri)
-                    .into(new SimpleTarget<Bitmap>() {
-                        @Override
-                        public void onResourceReady(Bitmap resource, Transition<? super Bitmap> transition) {
-                            if (DimenUtils.iconTooBlurry(holder.img.getResources(), resource.getWidth())) {
-                                defaultSetImageViewWithBackground(holder.img, site.getUrl(), FavIconUtils.getDominantColor(resource));
-                            } else {
-                                setImageViewWithBackground(holder.img, resource);
-                            }
-                        }
-                    });
+            // Tried AsyncTask and other simple offloading, the performance drops significantly.
+            // FIXME: 9/21/18 by saving bitmap color, cause FaviconUtils.getDominantColor runs slow.
+            final StrictMode.ThreadPolicy threadPolicy = StrictMode.allowThreadDiskWrites();
+            Bitmap resource = FavIconUtils.getBitmapFromUri(holder.itemView.getContext(), favIconUri);
+            if (DimenUtils.iconTooBlurry(holder.img.getResources(), resource.getWidth())) {
```

Comment:
I feel it's risky to do IO in UI thread.  Maybe there's a better way?

Reply:
I've tried simple AsyncTask and it's too slow and the UI seems janky with Animation. I'll file another bug to refactor this.

Reply:
I think we should modify the UX if it's compatible with current design.